manager/orchestrator/updater.go

```diff
@@ -135,18 +149,56 @@ func (u *Updater) Run(ctx context.Context, cluster *api.Cluster, service *api.Se
 		}()
 	}
 
-	for _, t := range dirtyTasks {
-		// Wait for a worker to pick up the task or abort the update, whichever comes first.
-		select {
-		case <-u.stopChan:
-			break
+	var failedTaskWatch chan events.Event
+
+	if service.Spec.Update == nil || service.Spec.Update.OnFailure == api.UpdateConfig_PAUSE {
```

Comment:
In that model, the failure will only be caught if the task never reaches `RUNNING`. The way it's done in this PR, any failure while the update is still running will pause the update. For example, a task can start up and then immediately die because it's misconfigured or the binary is missing or it can't connect to the database, etc.